### DISCUSSION TITLE
[Section request] AI/LLM

### DISCUSSION THREAD

**Discussion by C-Fu** [community-scripts/ProxmoxVE#755] [Request script]
### Application Name

Ollama, Stable Diffusion, LocalAI, gpt4all, etc

### Website

-

### Description

I believe a lot of the readily-available LXC-based scripts are already using (selfhosted) AI tools, so might as well we start having an AI section for it.

### Due Diligence

- [X] I have searched existing [scripts](https://community-scripts.github.io/Proxmox/scripts) and found no duplicates.
- [X] I have searched existing [discussions](https://github.com/community-scripts/ProxmoxVE/discussions?discussions_q=) and found no duplicate requests.
- [X] This is not a game-related request.

**Comment by MickLesk** [comment]
Thats are more then 1 Request.
Ollama und OpenWebUI are available

**Reply by michelroegl-brunner** [reply]
I think he is refering to the frontend/website the, not the install Scripts. I guess he is taking about a "AI" Category. 
I think this is not a bad idea. @BramSuurdje (Maybe you can have a look at it)

**Reply by MickLesk** [reply]
When we have enough Scripts yes

**Comment by darknessblade** [comment]
For things like Stable-Diffusion, I do think there need to be multiple helper-scripts [or force users to go trough the manual setup script]

since not everyone runs it on supported hardware. since for stable-diffusion a NVIDIA GPU is recommended.
There is a AMD fork, but its quite broken and doesn't work well with certain features like upscaling
[at least under windows], due unclear or incomplete documentation that is outdated.

For Stable-diffusion, the helper-script should not have a "AUTO-install" option, 
It should force people to go trough the manual prompt where they need to specify their hardware specifications for SD/ the AI tools they want to install.
It should also give ample warnings if unsupported hardware is detected.

A example for me would be: I have proxmox running on a Intel N100. and would be able to run a StableDiffusion fork called "easydiffusion" in a LXC container,
The IT/S won't be as high as with GPU rendering. but it should still be more than enough for simple Image generation.

**Comment by spijkey87** [comment]
I would like this
